bin/main.rb

```diff
@@ -1,3 +1,56 @@
 #!/usr/bin/env ruby
 
-puts 'Hello World!'
+def tic_tac_toe
+  intro
+  player_x = get_player('first')
+  player_o = get_player('second')
+  puts "\nNow lets start playing"
+  starter = start_bid(player_x, player_o)
+  puts "\n#{starter} wins bid to start"
+  if starter == player_x
+    play_game(player_x, player_o)
+  else
+    play_game(player_o, player_x)
+  end
+end
+
+def intro
+  puts "Hello and welcome to the world TIC-TAC-TOE\n\n"
+  puts "   |   |   \n --+---+--"
+  puts "   |   |   \n --+---+--"
+  puts "   |   |   \n "
+  puts "\nType in your names to start this awesome game\n"
+end
+
+def get_player(position)
+  puts "\nEnter name of #{position} Player"
+  player = gets.chomp
+  puts "\nWelcome #{player}"
+  player
+end
+
+def start_bid(player_x, player_o)
+  if rand(0..1).zero?
+    player_x
+  else
+    player_o
+  end
+end
+
+def play_game(first, second)
+  make_move(first)
+  make_move(second)
```

Comment:
**Hint:** If you call two turns in one single run of the loop if the first winner makes a winner the second player will still be asked to play. But the right behavior should be to finish the game once you detect a winner.